### PR TITLE
Add configuration warning when ButtonGroup is used with non-toggleable buttons

### LINKED ISSUE #70326: ButtonGroup doesn't emit signal
### Godot version

v4.0.beta8.official [45cac42c0]

### System information

Ubuntu 22.10, GTX1060 Vulkan

### Issue description

tried to connect multiple buttons with `ButtonGroup`, but unfortunately `ButtonGroup` doesn't emit `pressed` signal

```gdscript
extends Node


func _ready() -> void:
	var group := ButtonGroup.new()
#	var group : ButtonGroup = load("res://new_button_group.tres")  # this also doesn't work

	for c in get_children():
		c.button_group = group
	group.connect("pressed", _on_ButtonGroup_pressed)


func _on_ButtonGroup_pressed(button: BaseButton) -> void:
	print(1)
```

### Steps to reproduce

1. run attachment
2. press any button to printout

### Minimal reproduction project

[New Game Project.zip](https://github.com/godotengine/godot/files/10263969/New.Game.Project.zip)

### PR DESCRIPTION
**Pull request by Sauermann**
Also documentation of ButtonGroup was ambiguous.

resolve #70326

### CODE REVIEW

File: scene/gui/base_button.cpp [hunk 1]
```diff
@@ -374,6 +374,12 @@ void BaseButton::set_button_group(const Ref<ButtonGroup> &p_group) {
 		button_group->buttons.erase(this);
 	}
 
+#ifdef DEBUG_ENABLED
+	if (!toggle_mode) {
+		WARN_PRINT("ButtonGroup should only be used with buttons that have toggle_mode set to true.");
+	}
+#endif // DEBUG_ENABLED
+
```
[the remark this answers is not in the document]
Comment (Sauermann) [reply scene/gui/base_button.cpp:382]: `get_configuration_warnings` only covers the editor-UI, but not the case, where it is set via script. This warning covers both cases.
In the use-case of the linked issue the problem was with setting it via script, so i wanted to make sure, that there is at least a warning in that case.

[the remark this answers is not in the document]
Comment (akien-mga) [reply scene/gui/base_button.cpp:382]: The same is true for all configuration warnigns which usually match situations which can be reproduced via code. The current status quo is that we provide all the convenience of warnings in the editor without bloating the core API with warnings at runtime.

Some users will likely miss it, but configuration warning in editor + clear documentation should be a sufficient best effort in most cases.

[the remark this answers is not in the document]
Comment (Sauermann) [reply scene/gui/base_button.cpp:382]: Thanks for providing this guideline. I will adjust the PR accordingly.

[the remark this answers is not in the document]
Comment (Sauermann) [reply scene/gui/base_button.cpp]: I have updated the PR to use editor configuration_warnings.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**